### ISSUE TITLE
[FR] what feature asic need in 2022

### ISSUE THREAD

**Issue by kalcohol** [Oneflow-Inc/oneflow#7212]
For training:
[x] device abstract/isolated layer, so none cuda gpgpu or other training use asic can be easily wrapped
[ ] pytorch/tensorflow 2.0 pixel level python frontend
For inference:
[x] libtorch like c/c++ interface, for asic tools QAT or advanced PTQ algorithm which needs gradient descent

简而言之，首先需要 libtorch 类的 c/c++ 接口；
对于训练倾向的芯片设计，需要 CPU + ASIC IMPL 这样的后端组合，也就是提供硬件抽象层，方便进行 ASIC 后端对接；
对于 ASIC 类的推理芯片，需要有 hack 机制注册一个量化的替代 OP，进行量化过程数值模拟(含 Bit 位对齐)或梯度下降跑有梯度的 PTQ 算法；量化过的模型不理想时，可以导出量化模型进行 QAT；

如果力有所及，像素级对齐(或直接挪用)流行框架的前端。
